### PR TITLE
[doc] conf: Remove deprecated options from docs, Other cleanup

### PR DESCRIPTION
**Pull request by maflcko**
Some dumb fixes, like removing the mention that free transactions are still a thing or that rpcuser/pass should be used (as opposed to rpcauth or rpc cookie).

Combined with other fixes because I don't want to create 3 pull requests:
* conf: Remove deprecated options from docs
* Remove only mention of MIT/X11
* Link to developer notes in README.md

### CODE REVIEW

File: doc/build-osx.md [hunk 1]
```diff
@@ -80,9 +80,9 @@ Running
 
 Bitcoin Core is now available at `./src/bitcoind`
 
-Before running, it's recommended that you create an RPC configuration file:
+Before running, you may create an RPC configuration file:
```
Comment (practicalswift) [inline doc/build-osx.md]: No longer a RPC configuration file if empty? :-)

File: doc/build-osx.md [hunk 2]
```diff
@@ -80,9 +80,9 @@ Running
...
 
-    echo -e "rpcuser=bitcoinrpc\nrpcpassword=$(xxd -l 16 -p /dev/urandom)" > "/Users/${USER}/Library/Application Support/Bitcoin/bitcoin.conf"
+    echo > "/Users/${USER}/Library/Application Support/Bitcoin/bitcoin.conf"
```
Comment (practicalswift) [inline doc/build-osx.md]: Use `touch "/Users/${USER}/Library/Application Support/Bitcoin/bitcoin.conf"` instead

File: share/examples/bitcoin.conf [hunk 3]
```diff
@@ -71,12 +71,9 @@
 # is .cookie and found in the `-datadir` being used for bitcoind. This option is typically used
 # when the server and client are run as the same user.
 #
-# If not, you must set rpcuser and rpcpassword to secure the JSON-RPC api. The first
-# method(DEPRECATED) is to set this pair for the server and client:
-#rpcuser=Ulysseys
-#rpcpassword=YourSuperGreatPasswordNumber_DO_NOT_USE_THIS_OR_YOU_WILL_GET_ROBBED_385593
+# If not, you must set rpcuser and rpcpassword to secure the JSON-RPC api.
```
Comment (practicalswift) [inline share/examples/bitcoin.conf]: Uppercase API :-)

**Comment by practicalswift** [comment]
utACK fac57360d0f1375d8df34c57063d83e30cb5499f modulo nits :-)

File: doc/build-osx.md [hunk 4]
```diff
@@ -80,9 +80,9 @@ Running
 
 Bitcoin Core is now available at `./src/bitcoind`
 
-Before running, it's recommended that you create an RPC configuration file:
+Before running, you may create an empty RPC configuration file:
```
Comment (practicalswift) [inline doc/build-osx.md]: Do we generally refer to `bitcoin.conf` as "configuration file" or "RPC configuration file"?

**Comment by practicalswift** [comment]
ACK fa4da3c0588244e13bad66a84e1393fdf4735a81

**Comment by fanquake** [comment]
utACK fa4da3c

**Comment by hebasto** [comment]
utACK fa4da3c0588244e13bad66a84e1393fdf4735a81

**Comment by promag** [comment]
utACK fa4da3c.

**Pull request merged**